### PR TITLE
drivers: sensor: ams_iAQcore: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by galak**
Move driver to use DEVICE_DT_GET for handling access to bus parent
device struct.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

**Comment by galak** [comment]
Was commented on discord about use of `i2c_dt_spec`.  Not sure if it makes sense here or not, since the I2C address is hard coded.  Thoughts?

**Comment by gmarull** [comment]
> Was commented on discord about use of `i2c_dt_spec`. Not sure if it makes sense here or not, since the I2C address is hard coded. Thoughts?

I think we should (I2C address should come from DT), and use driver's config instead of data as well.

**Comment by galak** [comment]
> > Was commented on discord about use of `i2c_dt_spec`. Not sure if it makes sense here or not, since the I2C address is hard coded. Thoughts?
> 
> I think we should (I2C address should come from DT), and use driver's config instead of data as well.

Will like to see @MaureenHelm chime in and will update accordingly.

**Comment by MaureenHelm** [comment]
> > > Was commented on discord about use of `i2c_dt_spec`. Not sure if it makes sense here or not, since the I2C address is hard coded. Thoughts?
> > 
> > 
> > I think we should (I2C address should come from DT), and use driver's config instead of data as well.
> 
> Will like to see @MaureenHelm chime in and will update accordingly.

Yes, I think it makes sense here.

**Review by alexanderwachter — DISMISSED** [review]
👌

**Review by alexanderwachter — DISMISSED** [review]

File: drivers/sensor/ams_iAQcore/iAQcore.c [hunk 1]
```diff
@@ -114,6 +112,10 @@ static int iaq_core_init(const struct device *dev)
 
 static struct iaq_core_data iaq_core_driver;
 
+static struct iaq_core_config iaq_core_config = {
```
Comment (gmarull) [inline drivers/sensor/ams_iAQcore/iAQcore.c:115]:
```suggestion
static const struct iaq_core_config iaq_core_config = {
```

File: drivers/sensor/ams_iAQcore/iAQcore.h [hunk 2]
```diff
@@ -17,8 +17,11 @@ struct iaq_registers {
 	uint16_t voc;
 } __packed;
 
+struct iaq_core_config {
+	const struct i2c_dt_spec i2c;
```
Comment (gmarull) [inline drivers/sensor/ams_iAQcore/iAQcore.h:21]:
```suggestion
	struct i2c_dt_spec i2c;
```

**Review by gmarull — CHANGES_REQUESTED** [review]

**Review by gmarull — DISMISSED** [review]

File: drivers/sensor/ams_iAQcore/iAQcore.c [hunk 3]
```diff
@@ -100,12 +100,10 @@ static const struct sensor_driver_api iaq_core_driver_api = {
...
-	struct iaq_core_data *drv_data = dev->data;
+	const struct iaq_core_config *config = dev->config;
 
-	drv_data->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
-	if (drv_data->i2c == NULL) {
-		LOG_ERR("Failed to get pointer to %s device!",
-			    DT_INST_BUS_LABEL(0));
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("Bus device is not ready");
 		return -EINVAL;
```
Comment (MaureenHelm) [inline drivers/sensor/ams_iAQcore/iAQcore.c:107]: Return value should be changed to `-ENODEV`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**